### PR TITLE
Set player error on request failures

### PR DESCRIPTION
**Pull request by kimurakenshi**
# Summary

Hi @chemoish, currently We are using this package in our project and it is being really awesome so far. While debugging a concurrency restriction we  have in the backend I was trying to add a custom error message in our player to handle that scenario and noticed that for some http responses that come from our backend  the error is not being set in the player:

```javascript
this.player_.error({
  code: 0,
  message: errorMessage,
});
```
By looking at your code I think I found the problem so this PR is an attempt to fix it.
When we  
```javascript 
const request = new XMLHttpRequest(); 
```
and then

```javascript 
    request.addEventListener('error', this.onCertificateError, false);
```
the only errors that `error` listen to are network errors and cross-origin errors, when I http response comes from the backend `load` handler will be called instead and the request is considered successful.
So, in order to catch all the errors that comes from the backend we have to check the satus code of the request and when is not `200` execute the error logic that will end up setting a custom error in the player instance.

Also I found a dependency error while trying to `npm run build` for `babel-core` so I added that package to `devDependencies` in a different commit. That change is totally unrelated to this PR so let me know if you want to remove that commit from this PR.

Feedback is more than welcome!

Thank you.

### CODE REVIEW

File: src/videojs-fairplay.js [hunk 1]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
...
+      Response is ${this.getErrorResponse(request.response)}`;
+
+    this.player_.error({
+      code: 0,
+      message: errorMessage,
+    });
+  }
+
+  getErrorResponse(response) {
+    return response ?
```
Comment (chemoish) [inline src/videojs-fairplay.js]:
```
if (!response) {
  return 'NONE';
}

return <String…>;
```

File: src/videojs-fairplay.js [hunk 2]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
...
     callback();
   }
 
+  onRequestError(request, errorType = ERROR_TYPE.UNKNOWN) {
+    this.log('onRequestError()');
+
+    const errorMessage = `${errorType} - DRM: com.apple.fps.1_0 update, 
+      XHR status is '${request.statusText}(${request.status})', expected to be 200. 
+      readyState is '${request.readyState}'. 
+      Response is ${this.getErrorResponse(request.response)}`;
```
Comment (chemoish) [inline src/videojs-fairplay.js]: Should the response section also be wrapped by `'${this.getErrorResponse}'.`?

File: src/videojs-fairplay.js [hunk 3]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
...
+
+    certificate = new Uint8Array(response);
 
     callback();
   }
 
+  onRequestError(request, errorType = ERROR_TYPE.UNKNOWN) {
+    this.log('onRequestError()');
+
+    const errorMessage = `${errorType} - DRM: com.apple.fps.1_0 update,
```
Comment (chemoish) [inline src/videojs-fairplay.js:159]: Does this error message get properly formatted or does it render weird due to the ```?

File: src/videojs-fairplay.js [hunk 4]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
...
+      readyState is '${request.readyState}'. 
+      Response is ${this.getErrorResponse(request.response)}`;
+
+    this.player_.error({
+      code: 0,
+      message: errorMessage,
+    });
+  }
+
+  getErrorResponse(response) {
```
Comment (chemoish) [inline src/videojs-fairplay.js]: alpha please.

File: src/videojs-fairplay.js (hunk 3 above)
Comment (chemoish) [inline src/videojs-fairplay.js:159]: Error structure matches current dash specification, seems fine.

https://github.com/Dash-Industry-Forum/dash.js/blob/be4350883185f22b5f50c3ad6e29c0402c70d584/src/streaming/protection/controllers/ProtectionController.js#L497

File: src/videojs-fairplay.js [hunk 5]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
...
+    return response ?
+      String.fromCharCode.apply(null, new Uint8Array(response)) :
+      'NONE';
```
Comment (chemoish) [inline src/videojs-fairplay.js]: https://github.com/Dash-Industry-Forum/dash.js/blob/be4350883185f22b5f50c3ad6e29c0402c70d584/src/streaming/protection/controllers/ProtectionController.js#L497

NONE is what they respond with, i guess is fine?

File: src/videojs-fairplay.js (hunk 3 above)
Comment (kimurakenshi) [reply src/videojs-fairplay.js:159]: I think so: 
screen shot 2016-12-22 at 5 40 40 pm

File: src/videojs-fairplay.js [hunk 6]
```diff
@@ -128,11 +129,42 @@ class Html5Fairplay {
   onCertificateLoad(event, { callback }) {
     this.log('onCertificateLoad()');
 
-    certificate = new Uint8Array(event.target.response);
+    const {
+      response,
+      status,
+    } = event.target;
+
```
Comment (chemoish) [inline src/videojs-fairplay.js:144]: wondering if this is an issue, not sure if response will always be binary, can fix if error later.

**Comment by chemoish** [comment]
@kimurakenshi Thanks for your gracious help!

**Pull request merged**